content/docs/atom-guide/dev-guide/materials/lua-material-functor-api.md

```diff
@@ -0,0 +1,316 @@
+  * ClearDepthWriteMask
+  * **SetDepthComparisonFunc**([`ComparisonFunc`](#comparisonfunc))
+  * ClearDepthComparisonFunc
+  * **SetStencilEnabled**(`boolean`)
+  * ClearStencilEnabled
+  * **SetStencilReadMask**(`number`)
+  * ClearStencilReadMask
+  * **SetStencilWriteMask**(`number`)
+  * ClearStencilWriteMask
+  * **SetStencilFrontFaceFailOp**([`StencilOp`](#stencilop))
+  * ClearStencilFrontFaceFailOp
+  * **SetStencilFrontFaceDepthFailOp**([`StencilOp`](#stencilop))
+  * ClearStencilFrontFaceDepthFailOp
+  * **SetStencilFrontFacePassOp**([`StencilOp`](#stencilop))
+  * ClearStencilFrontFacePassOp
+  * **SetStencilFrontFaceFunc**([`ComparisonFunc`](#comparisonfunc))
+  * ClearStencilFrontFaceFunc
+  * **SetStencilBackFaceFailOp**([`StencilOp`](#stencilop))
+  * ClearStencilBackFaceFailOp
+  * **SetStencilBackFaceDepthFailOp**([`StencilOp`](#stencilop))
+  * ClearStencilBackFaceDepthFailOp
+  * **SetStencilBackFacePassOp**([`StencilOp`](#stencilop))
+  * ClearStencilBackFacePassOp
+  * **SetStencilBackFaceFunc**([`ComparisonFunc`](#comparisonfunc))
+  * ClearStencilBackFaceFunc
+  
+## Enum Types
+  
+These enum types are reflected to lua with the following global values:
+
+#### MaterialPropertyVisibility
+
+* MaterialPropertyVisibility_Enabled
+* MaterialPropertyVisibility_Disabled
+* MaterialPropertyVisibility_Hidden
+
+#### FillModef
+
+* FillMode_Solid
+* FillMode_Wireframe
+* FillMode_Invalid
+
+#### CullMode
+
+* CullMode_None
+* CullMode_Front
+* CullMode_Back
+* CullMode_Invalid
+
+#### ComparisonFunc
+
+{{< note >}}
+These enum values might not actually be reflected as intended, it seems to be an oversight. See https://github.com/o3de/o3de/issues/10203
+{{< /note >}}
+
+* ComparisonFunc_Never
+* ComparisonFunc_Less
+* ComparisonFunc_Equal
+* ComparisonFunc_LessEqual
+* ComparisonFunc_Greater
+* ComparisonFunc_NotEqual
+* ComparisonFunc_GreaterEqual
+* ComparisonFunc_Always
+* ComparisonFunc_Invalid
+
+#### BlendFactor
+
+* BlendFactor_Zero
+* BlendFactor_One
+* BlendFactor_ColorSource
+* BlendFactor_ColorSourceInverse
+* BlendFactor_AlphaSource
+* BlendFactor_AlphaSourceInverse
+* BlendFactor_AlphaDest
+* BlendFactor_AlphaDestInverse
+* BlendFactor_ColorDest
+* BlendFactor_ColorDestInverse
+* BlendFactor_AlphaSourceSaturate
+* BlendFactor_Factor
+* BlendFactor_FactorInverse
+* BlendFactor_ColorSource1
+* BlendFactor_ColorSource1Inverse
+* BlendFactor_AlphaSource1
+* BlendFactor_AlphaSource1Inverse
+* BlendFactor_Invalid
+
+#### BlendOp
+
+* BlendOp_Add
+* BlendOp_Subtract
+* BlendOp_SubtractReverse
+* BlendOp_Minimum
+* BlendOp_Maximum
+* BlendOp_Invalid
+
+#### DepthWriteMask
+
+* DepthWriteMask_Zero
+* DepthWriteMask_All
+* DepthWriteMask_Invalid
+
+#### StencilOp
+
+* StencilOp_Keep
+* StencilOp_Zero
+* StencilOp_Replace
+* StencilOp_IncrementSaturate
+* StencilOp_DecrementSaturate
+* StencilOp_Invert
+* StencilOp_Increment
+* StencilOp_Decrement
+* StencilOp_Invalid
+
+
```

Comment:
Create a related topics section. Add a link to the "Lua" section in User Guide.

```
## Related topics
| Related topics | Details | 
| --- | --- |
| | [Lua](/docs/user-guide/scripting/lua/) | Use Lua in Open 3D Engine (O3DE) to automate your game project. |
```

Reply:
I believe you are referring to https://www.o3de.org/docs/user-guide/scripting/lua/ . This section of documentation is dedicated to lua gameplay scripting in O3DE. This is a separate topic and mostly unrelated to lua material functors. It's the same scripting language but the APIs are totally different. The only part that would be relevant are the links to official lua documentation (https://www.o3de.org/docs/user-guide/scripting/lua/#learning-lua).

I can add a note to clarify this.

Reply:
In that case, a link to the Lua documentation would be helpful! Can you still add it to a Related topics section at the bottom of this page?